build/tasks/node_smoke_tests.js

```diff
@@ -17,33 +18,116 @@ if ( !verifyNodeVersion() ) {
 // important so that the tests & the main process don't interfere with
 // each other, e.g. so that they don't share the `require` cache.
 
-async function runTests( sourceType, module ) {
-	if ( !allowedModules.includes( sourceType ) ) {
-		throw new Error(
-			`Usage: \`node_smoke_tests [${allowedModules.join( "|" )}]:JQUERY\``
-		);
+async function runTests( { libraryType, sourceType, module } ) {
+	if ( !allowedLibraryTypes.includes( libraryType ) ||
+			!allowedSourceTypes.includes( sourceType ) ) {
+		throw new Error( `Incorrect libraryType or sourceType value; passed: ${
+			libraryType
+		} ${ sourceType } "${ module }"` );
 	}
-	const dir = `./test/node_smoke_tests/${sourceType}`;
+	const dir = `./test/node_smoke_tests/${ sourceType }/${ libraryType }`;
 	const files = await fs.promises.readdir( dir, { withFileTypes: true } );
 	const testFiles = files.filter( ( testFilePath ) => testFilePath.isFile() );
+
+	if ( !testFiles.length ) {
+		throw new Error( `No test files found for ${
+			libraryType
+		} ${ sourceType } "${ module }"` );
+	}
```

Comment:
When making the changes after the rebase, I noticed tests are finishing a bit too quickly. Then I realized I forgot to update the `dir` definition. I added some validation that anything got run.

It'd be a bit safer to also print a list of files with tests that are executed as then if wrong ones were run, I'd notice immediately. This would make the output here a bit longer, though, and we don't have a good mechanism for verbose mode so I didn't add anything like that.

@timmywil what do you think? Should we print the list of test files?

Reply:
I don't think so. We probably won't touch this file much in the future and I'd prefer less noise. We could add more checks if it becomes an issue.

Reply:
Cool, so what we have here should be enough then.